### PR TITLE
Add METHOD target for use with AutoValue's abstract property methods

### LINKED ISSUE #687: Add METHOD Target to SerializedName attribute
I'm in the process of making an AutoValue extension that will generate a Gson TypeAdapterFactory for generated data types, and I'd like to support `@SerializedName`.  Currently, the attribute only targets `ElementType.FIELD`s, meaning users can't put it on the abstract property methods that AutoValue uses.  I'd like to have `ElementType.METHOD` added to the list of supported `ElementType`s so that I don't have to include a separate annotation for AutoValue users.

Here is what the user's class would look like in this case:

```java
@AutoValue public abstract class Foo {
  @SerializedName("Baz") public abstract String baz();
}
```

### PR DESCRIPTION
**Pull request by MichaelEvans**
Closes #687

### CODE REVIEW

File: gson/src/main/java/com/google/gson/annotations/SerializedName.java [hunk 1]
```diff
@@ -64,7 +64,7 @@
  * @author Joel Leitch
  */
 @Retention(RetentionPolicy.RUNTIME)
-@Target(ElementType.FIELD)
+@Target({ElementType.FIELD, ElementType.METHOD})
```
Comment (inder123) [inline gson/src/main/java/com/google/gson/annotations/SerializedName.java]: @swankjesse Do we really want to do this?

If we allow SerializedName on methods, we should actually support it in full. 
We Don't and we seem to have no plans to support it either. So, why open it up?

**Comment by swankjesse** [comment]
It's a pragmatic solution to the problem AutoValue users have. I don't think it obligates us to doing getters and setters, and I'm quite against doing that!

**Comment by swankjesse** [comment]
Details are here: https://github.com/google/gson/issues/687

**Comment by inder123** [comment]
Ok

**Pull request merged**